jmxtrans-core/src/main/java/com/googlecode/jmxtrans/model/Server.java

```diff
@@ -148,10 +155,21 @@ public String getAlias() {
 		return this.alias;
 	}
 
+	/**
+	 * Returns the pid of the local process jmxtrans will attach to
+	 *
+	 * @return pid of the java process
+	 */
+	public String getPid() {
+		if(this.pid == null) {
+			return null;
+		}
+		return this.pid;
+	}
+
 	public String getHost() {
 		if (host == null && url == null) {
-			// TODO: shouldn't we just return a null in this case ?
-			throw new IllegalStateException("host is null and url is null. Cannot construct host dynamically.");
+			return null;
```

Comment:
I really dislike methods returning null (there are a lot of them for "historical reasons"). Still, let's try not to introduce new ones. In the case of a local JVM accessed by PID, the hostname of the JVM is the hostname of the machine on which JmxTrans runs. So we have a way to cleanly expose a hostname also in this case. So let's not break the contract unless we really have to.